### DISCUSSION TITLE
对@formily/designable-antd代码设计的几个疑问

### DISCUSSION THREAD

**Discussion by jinphic** [alibaba/designable#211] [Q&A]
然后我在准备将步骤器(FormStep)和走马灯(Carousel) 提个mr的时候，很疑问，为什么代码的设计结构里面只有 preview.tsx一个文件。在我看来，design和preview两个状态是有很多不同，在design太可以放过一下状态的判断，比如required，比如走马灯的autoplay。

**Comment by janryWang** [comment]
什么意思？

**Comment by hlzzd666** [comment]
design和preview 本来就可以是不相关的吧，你可以设计两套组件文件呀，一套用在design，一套用在preview，在design中生成jsonSchema，然后拿去在preview渲染。个人理解，有问题的话不要骂我。
